### ISSUE TITLE
Road and POI labels text cannot display when switch endpoint to .cn

### ISSUE THREAD

**Issue by m-stephen** [mapbox/mapbox-navigation-ios#1554]
Now I'm focus on navigation project inside China by using our navigation-ios SDK. When I switch the endpoint to .cn, all the roads and POI labels text cannot display:

![image](https://user-images.githubusercontent.com/2045318/42561679-52789cac-852c-11e8-8f9b-d863c63166b9.png)

After deep dive into this bug, I found that during the `func mapView(_ mapView: MGLMapView, didFinishLoading style: MGLStyle)` function in `NavigationMapView.swift`, our navigation SDK do a localized logic called `self.mapView.localizeLabels()`.

In this function, I saw a couple of lines code:

            let localizedText = text.mgl_expressionLocalized(into: locale)
            if localizedText != text {
                layer.text = localizedText
            }

This function make the labels can be localized to the phone's region & language.
However, in .cn map, layers' properties is a little bit different from global side:

China layers sample:
![image](https://user-images.githubusercontent.com/2045318/42562423-446502fc-852e-11e8-8abb-3403e9bbee9c.png)

This is our global [layer properties' link](https://www.mapbox.com/studio/tilesets/mapbox.mapbox-streets-v7/)

As you see, the layer's from .cn only have the `name_zh` but in `text.mgl_expressionLocalized(into: locale)`, it return the `name_zh-Hans`. I know this function is in our map SDK but I'm not very sure it should be fixed in client side or server side.

After talking with @zmofei, he said the .cn data is directly provided by EMG and it's hard to modify in a short term, so I think fix on client side is a better choice. Should we do some downgrade strategy if our layers do not contain `name_zh-Hans` property? @1ec5 @bsudekum 

cc: @chriswu42 @tony-cj

**Comment by 1ec5** [comment]
I was unaware that China Streets differs from the rest-of-world Streets source in this way. Are there any other differences that we should be aware of? The navigation SDK makes a lot of assumptions about the Streets source’s fields and layers.

For this particular issue, we should introduce a special case in the map SDK that varies [these lists of locale identifiers](https://github.com/mapbox/mapbox-gl-native/blob/bf69f1a2d3ed4e6792b5ebc704b9cb70c02e6f57/platform/darwin/src/MGLVectorTileSource.mm#L78-L96) for the .cn API endpoint. The fix will complicate mapbox/mapbox-gl-native#11867 a bit, but I think that’s OK. Would you mind opening an issue about this, along with a link to the public documentation for this source, if it’s available? Thanks!

**Comment by m-stephen** [comment]
Just ask @zmofei to give a China street source [link](https://www.mapbox.com/studio/tilesets/zhuwenlong.ctw8q96o/). 

Do you mean I should open a same ticket in mapbox-gl-native? @1ec5

**Comment by 1ec5** [comment]
Yes, we should track this issue in mapbox-gl-native, since a proper fix would go into the map SDK.

In the meantime, we can also work around the issue in the navigaiton SDK. The code below currently uses the `mul` locale identifier for roads and `nil` for everything else. `mul` should result in Chinese in China, but as long as .cn is in use, we’d need to replace `nil` with the `zh-CN` locale:

https://github.com/mapbox/mapbox-navigation-ios/blob/e5f371538d6eef4623828061c5509f2f7a8003fc/MapboxNavigation/NavigationMapView.swift#L971-L972

**Comment by m-stephen** [comment]
After thinking of our navigation SDK for supporting .cn map, and I think there should be a .cn network module like our map SDK(Refer to MGLNetworkConfiguration.h & MGLNetworkConfiguration.m).

It can make our SDK much easier to switch the endpoint: not only the map endpoint, also the api-direction, tts services endpoint.

For deep diving into mgl_expressionLocalizedIntoLocale function in our map SDK, I think it's better to do nothing in localizeLabels function when the map is switch the endpoint to .cn.

PR is [here](https://github.com/mapbox/mapbox-navigation-ios/pull/1558). @1ec5

**Comment by 1ec5** [comment]
> a proper fix would go into the map SDK

mapbox/mapbox-gl-native#12387 introduces a fallback from `name_zh-Hans` to `name`. I’ve proposed that it also fall back, opportunistically, from `name_zh-Hans` to `name_zh-CN` to `name`.
